### PR TITLE
[mesheryctl] Implement meshery update prompt mesheryctl

### LINKED ISSUE #1644: [mesheryctl] Prompt user to update to latest Meshery server release and perform update
**Current Behavior**
`mesheryctl version` will notify users of a later server version available and provides the user with instructions to do so, but does not update to the latest version on the user's behalf.

**Desired Behavior**
`mesheryctl` should perform the upgrade to the latest version of Meshery server on behalf of the user, if user so chooses. This update will be for Meshery server and Meshery adapters.

**Implementation**
`mesheryctl version` should call to the internal function/command that is equivalent to `mesheryctl system update`

**Mockups**
`mesheryctl version` should prompt the user with this question:

```
Update Meshery server and Meshery adapters to the latest available versions? (y/n):
```

**Acceptance Tests**
1. If later version of Meshery server is available and the user responds with "y", Meshery should automatically be updated.
1. Users should be able to invoke this operation for Meshery deployments on Docker.
1. Users should be able to invoke this operation for Meshery deployments on Kubernetes.
1. `mesheryctl` will not be updated.

---
**Additional Context**

> I have looked at the code but did not test it. However, I do have some suggestions. I think when mesheryctl is run with or without a parameter, the parameter should be saved and the version should be checked at that time and printed. The user should then be informed and should be prompted to run update at that time. If the user chooses not to update, then the parameter should be used to execute the requested command, else mesheryctl should invoke the update command and then execute the originally requested command by using the saved parameter.

Great suggestion, @vijcheru. I agree. In a subsequent PR, let's prompt the user with "Would you like to upgrade to vx.x.x now? (y/n)?" and process accordingly.

_Originally posted by @leecalcote in https://github.com/layer5io/meshery/pull/1640#issuecomment-703320153_

### PR DESCRIPTION
**Pull request by uds5501**
Signed-off-by: Uddeshya Singh <singhuddeshyaofficial@gmail.com>

**Description**
I have implemented a function essentially similar to `updateMesheryContainers()` (in `update.go`)
I initially tried to call that function directly in the handler but since it's nested under `internal/`, I can't, so pardon the code duplicacy.

This PR fixes #1644 

**Notes for Reviewers**
A sample run in my local system gave:
```cmd

Client Version: 0.4.24 	  GitSHA: 8a95887a4c4ff9681d8c27ad8d3a83840f8524c8
Server Version: v0.4.25 	  GitSHA: 6b23b79
Checking for latest version of Meshery....

0.4.24 is not the latest Meshery release. Update to v0.4.25. Run `mesheryctl system update`

Would you like to upgrade to v0.4.25 now? (y/n)?
y
Updating Meshery now...
Pulling meshery          ... done
Pulling meshery-istio    ... done
Pulling meshery-linkerd  ... done
Pulling meshery-consul   ... done
Pulling meshery-octarine ... done
Pulling meshery-nsm      ... done
Pulling meshery-kuma     ... done
Pulling meshery-cpx      ... done
Pulling meshery-osm      ... done
Pulling meshery-nginx-sm ... done
Pulling watchtower       ... done
```

**[Signed commits](https://github.com/layer5io/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by uds5501** [comment]
@kushthedude could you take a look?

**Comment by kushthedude** [comment]
```
I have implemented a function essentially similar to updateMesheryContainers() (in update.go)
I initially tried to call that function directly in the handler but since it's nested under internal/, I can't, so pardon the code duplicacy.
```
Hey @uds5501 , What we can do over here is move the updateContainers function into `mesheryctl/pkg/utils/helpers.go` and from there both updateCmd and the handlers can import 😄

**Comment by uds5501** [comment]
> ```
> I have implemented a function essentially similar to updateMesheryContainers() (in update.go)
> I initially tried to call that function directly in the handler but since it's nested under internal/, I can't, so pardon the code duplicacy.
> ```
> 
> Hey @uds5501 , What we can do over here is move the updateContainers function into `mesheryctl/pkg/utils/helpers.go` and from there both updateCmd and the handlers can import

That sounds good, I'll update the PR accordingly

**Comment by uds5501** [comment]
@kushthedude take a look now?

**Review by kushthedude — APPROVED** [review]
lgtm!

**Comment by uds5501** [comment]
@leecalcote @abhishekkumkar could you guys take a look?

File: handlers/server_spec_handler.go [hunk 1]
```diff
@@ -66,6 +68,15 @@ func CheckLatestVersion(serverVersion string) (*latest.CheckResponse, error) {
 	// If user is running an outdated release, let them know.
 	if res.Outdated {
 		logrus.Info("\n", serverVersion, " is not the latest Meshery release. Update to v", res.Current, ". Run `mesheryctl system update`")
+		logrus.Info("\nWould you like to upgrade to v", res.Current, " now? (y/n)?")
```
Comment (kumarabd) [inline handlers/server_spec_handler.go:71]: @uds5501 https://github.com/manifoldco/promptui - A library that implements clean prompt. Issue is that not just y but we should accept and empty `enter` or `yes` as an answer.

**Review by anirudhjain-fareye — APPROVED** [review]
Looks good to me too

**Review by kumarabd — CHANGES_REQUESTED** [review]
Could you address to replacing the prompts with a library? @uds5501

File: handlers/server_spec_handler.go (hunk 1 above)
Comment (uds5501) [reply handlers/server_spec_handler.go:71]: @kumarabd sounds good, will fix that!

Comment (leecalcote) [reply handlers/server_spec_handler.go:71]: Good work, @uds5501. Let’s use only one question mark.

**Comment by uds5501** [comment]
@kumarabd @leecalcote 
Completed the changes requested.
New prompt UI looks something like this and functionality works as expected

```cmd

0.4.24 is not the latest Meshery release. Update to v0.4.25. Run `mesheryctl system update`
Use the arrow keys to navigate: ↓ ↑ → ← 
? Would you like to upgrade to v0.4.25 now?: 
  ▸ Yes
    No
```

File: handlers/server_spec_handler.go [hunk 2]
```diff
@@ -66,6 +69,22 @@ func CheckLatestVersion(serverVersion string) (*latest.CheckResponse, error) {
 	// If user is running an outdated release, let them know.
 	if res.Outdated {
 		logrus.Info("\n", serverVersion, " is not the latest Meshery release. Update to v", res.Current, ". Run `mesheryctl system update`")
+		promptLabel := fmt.Sprintf("Would you like to upgrade to v%s now?", res.Current)
+
+		prompt := promptui.Select{
+			Label: promptLabel,
+			Items: []string{"Yes", "No"},
```
Comment (kumarabd) [inline handlers/server_spec_handler.go:76]: small please, `yes`

**Review by kumarabd — COMMENTED** [review]
Rest looks good

**Review by kumarabd — APPROVED** [review]

**Pull request merged**